pylint_odoo/test/main.py

```diff
@@ -432,6 +433,42 @@ def test_140_check_migrations_is_not_odoo_module(self):
         expected_errors = {}
         self.assertDictEqual(real_errors, expected_errors)
 
+    @unittest.skipUnless(
+        sys.version_info[0:2] >= (3, 6),
+        "Fstrings (PEP498) are only supported since 3.6"
+    )
+    def test_145_check_fstring_sqli(self):
+        """Verify the linter is capable of finding SQL Injection vulnerabilities
+        when using fstrings.
+        Related to https://github.com/OCA/pylint-odoo/issues/363"""
+        extra_params = [
+            '--disable=all',
+            '--enable=sql-injection'
+        ]
+        queries = '''
+            def fstring_sqli(self):
+                self.env.cr.execute(f"SELECT * FROM TABLE WHERE SQLI = {self.table}")
+                self.env.cr.execute(
+                    f"SELECT * FROM TABLE WHERE SQLI = {'hello' + self.name}"
+                )
+                self.env.cr.execute(f"SELECT * FROM {self.name} WHERE SQLI = {'hello'}")
+                death_wish = f"SELECT * FROM TABLE WHERE SQLI = {self.name}"
+                self.env.cr.execute(death_wish)
+            def fstring_no_sqli(self):
+                self.env.cr.execute(f"SELECT * FROM TABLE WHERE SQLI = {'hello'}")
+                self.env.cr.execute(
+                    f"CREATE VIEW {self._table} AS (SELECT * FROM res_partner)"
+                )
+                self.env.cr.execute(f"SELECT NAME FROM res_partner LIMIT 10")
+            '''
+        with NamedTemporaryFile(mode='w', encoding='utf-8', delete=False) as f:
+            self.addCleanup(os.remove, f.name)
+            f.write(dedent(queries).lstrip())
```

Comment:
What about?

```suggestion
        with NamedTemporaryFile(mode='w', encoding='utf-8') as f:
            f.write(queries)
            f.flush()
            pylint_res = self.run_pylint([f.name], extra_params)
```

You even can use the string as you need it without calling extra string formatters:
```python
        queries = '''
def fstring_sqli(self):
    self.env.cr.execute(f"SELECT * FROM TABLE WHERE SQLI = {self.table}")
    self.env.cr.execute(
        f"SELECT * FROM TABLE WHERE SQLI = {'hello' + self.name}"
    )
    self.env.cr.execute(f"SELECT * FROM {self.name} WHERE SQLI = {'hello'}")
    death_wish = f"SELECT * FROM TABLE WHERE SQLI = {self.name}"
    self.env.cr.execute(death_wish)
def fstring_no_sqli(self):
    self.env.cr.execute(f"SELECT * FROM TABLE WHERE SQLI = {'hello'}")
    self.env.cr.execute(
        f"CREATE VIEW {self._table} AS (SELECT * FROM res_partner)"
    )
    self.env.cr.execute(f"SELECT NAME FROM res_partner LIMIT 10")'''
```

Reply:
I used dedent so the code looked prettier (IMO). I'll make the changes.